ui/src/store/ui-slice.ts

```diff
@@ -0,0 +1,63 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+
+interface UiState {
+    themeKey: ThemeKey;
+    connectionErrorMessage: string | null;
+    navOpen: boolean;
+    showSettings: boolean;
+    snack: {
+        messages: string[];
+        message: string | null;
+    }
+}
+
+const initialUiState: UiState = {
+    themeKey: 'dark',
```

Comment:
I'm not sure I like this too, previously this was local component state, because it was only used in two places, and passed downwards to one component. Sure adding this to the global state can be done, but local component state is better as it's simpler and easier to navigate.

These refactorings making this review much more effort as it's not just library change / upgrades, but also logic changes which aren't 100% necessary.

Reply:
I had a lot of problems in the beginning to understand which prop is passed to which component and why
Using a global state reduces the need of props drilling which is making the componets match leaner and easier to read.

But if you prefer a global state in e.g. app.tsx, I can remove these UI related states from the global state and move it to app.tsx and pass them via props drilling to the components.
But to be honest it will be technically equal and will just make the components harder to read.

But as you prefer.

Reply:
I prefer it to be as close as possible to the original implementation for now, as it makes less changes to review and less possibility of bugs, because the code already runs for some time. This includes the moving of the selected plugin state into redux.

Reply:
ok, I will move the state/slice/action definition to the component.
But I will not remove the ui state management and will change everything back to prop drilling.
This is error prone and we will for sure just add new bugs by this. The UI state is working perfectly fine.

Reply:
> This is error prone and we will for sure just add new bugs by this. 

How is it error prone to pass properties to sub components. TypeScript is used, so it's not like you can miss passing the props?